Sources/JWTProvider/PayloadAuthenticationMiddleware.swift

```diff
@@ -64,22 +64,24 @@ public final class PayloadAuthenticationMiddleware<U: PayloadAuthenticatable>: M
 
         let filteredSigners = try self.signers(for: jwt)
 
-        // Try to use all the signers until one matches
-        var verified = false
+        // Default Error
+        var error: Swift.Error? = JWTError.signatureVerificationFailed
 
+        // Try to use all the signers until one matches
         for signer in filteredSigners {
 
             do {
-                _ = try req.jwt(verifyUsing: signer, and: claims)
-                verified = true
+                _ = try req.jwt(verifyUsing: signer, and: self.claims)
+                error = nil
                 break
-            } catch {
+            } catch let jwtError {
+                error = jwtError
                 continue
             }
         }
 
-        guard verified else {
-            throw JWTProviderError.noVerifiedJWT
```

Comment:
The current released version doesn't support `kid` at all, right? So we don't need to worry about backward compatibility.

I'd say we should require a kid iff there are multiple signers. If there's just one signer, then we don't need a kid and we just use the only available signer. 

Throwing away errors as this would require will very likely cause bugs. Also validating against multiple signers with each request could potentially be a performance issue.